lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1833,6 +1834,31 @@ sub test_resultfile_list {
     return \@filelist_existing;
 }
 
+=head2 investigate
+
+Find pointers for investigation on failures, e.g. what changed vs. a "last
+good" job in the same scenario.
+
+=cut
+sub investigate {
+    my ($self, %args) = @_;
+    my @previous = $self->_previous_scenario_jobs;
+    return {error => 'No previous job in this scenario, cannot provide hints'} unless @previous;
+    my %investigation;
+    return {error => 'No result directory available for current job'} unless $self->result_dir();
+    for my $prev (@previous) {
+        next unless ($prev->result =~ '(passed|softfailed)');
```

Comment:
This would look better as:
```perl
next unless $prev->result =~ /(?:passed|softfailed)/;
```
And it's more efficient not to capture if you don't have to.

Reply:
done